### PR TITLE
Feature/parth4apple/comments

### PR DESCRIPTION
**Pull request by parth4apple**
## Changes
- Created the comments and replies display based on the high fidelity designs
- Added styled component to dependency list

## To do
- like comment functionality
- reply connection to backend
- add a new comment field 
- new comment connection to backend

### CODE REVIEW

File: src/Restroom/RestroomCommentChain.js [hunk 1]
```diff
@@ -1,28 +1,199 @@
...
+  useEffect(() => {
+    axios.get("/restrooms/replies/" + props.id).then((response) => {
+      console.log(response.data);
+      setReplies(response.data);
+    });
+  }, []);
+
+  return (
+    <>
+      <CommentComponent
```
Comment (wes-chen) [inline src/Restroom/RestroomCommentChain.js:22]: why is this hard coded?

File: src/Restroom/RestroomCommentChain.js [hunk 2]
```diff
@@ -1,28 +1,199 @@
...
+          />
+        );
+      })}
+      <Reply />
+    </>
+  );
+}
+
+// individual comment component -- used for comments and replies
+function CommentComponent(props) {
```
Comment (wes-chen) [inline src/Restroom/RestroomCommentChain.js:45]: can this be moved out into a different file

File: src/Restroom/RestroomCommentChain.js [hunk 3]
```diff
@@ -1,28 +1,199 @@
...
+import poopEmojiImg from "../assets/brownPoop.png";
+import like from "../assets/like.png";
+import ratingPoop from "../assets/ratingPoop.png";
+
+export default function RestroomCommentChain(props) {
+  const [replies, setReplies] = useState([]);
+
+  useEffect(() => {
+    axios.get("/restrooms/replies/" + props.id).then((response) => {
+      console.log(response.data);
```
Comment (wes-chen) [inline src/Restroom/RestroomCommentChain.js:15]: we don't need to log this anymore

File: src/Restroom/RestroomCommentChain.js [hunk 4]
```diff
@@ -1,28 +1,199 @@
...
+        comment={props.body}
+        rating={4}
+      />
+      {replies.map((comment) => {
+        return (
+          <CommentComponent
+            key={comment._id}
+            username={comment.username}
+            comment={comment.body}
+            likes={4}
```
Comment (wes-chen) [inline src/Restroom/RestroomCommentChain.js:34]: is this information not available from the API?

File: src/Restroom/RestroomCommentChain.js [hunk 5]
```diff
@@ -1,28 +1,199 @@
...
+          <Heart src={like} onClick={() => handleLike()} />
+          <SmallText>{"(" + (props.likes || 0) + ")"}</SmallText>
+        </LikesContainer>
+      </CommentContainer>
+    </LineContainer>
+  );
 }
+
+// reply component -- used in CommentComponent
+function Reply() {
```
Comment (wes-chen) [inline src/Restroom/RestroomCommentChain.js:81]: different file like earlier

**Review by wes-chen — CHANGES_REQUESTED** [review]
few comments other than the ones we discussed on slack

File: src/Restroom/RestroomDiscussion.js [hunk 6]
```diff
@@ -1,29 +1,41 @@
...
   render() {
-    return(
+    return (
       <Card>
         <CardBody>
           <CardTitle tag="h5">Discussion Time.</CardTitle>
-          <CardSubtitle className="mb-2 text-muted" tag="h6">There are {this.props.num_comments} comments on this restroom. </CardSubtitle>
-          {this.state.comments.map(comment => <RestroomCommentChain key={comment._id} id={comment._id} username={comment.username} body={comment.body} />)}
+          <CardSubtitle className="mb-2 text-muted" tag="h6">
+            There are {this.props.num_comments} comments on this restroom.{" "}
```
Comment (wes-chen) [inline src/Restroom/RestroomDiscussion.js:28]: You can remove this, it was just a placeholder

File: src/Restroom/RestroomCommentChain.js [hunk 7]
```diff
@@ -1,28 +1,39 @@
...
-  }
+  useEffect(() => {
+    axios.get("/restrooms/replies/" + props.id).then((response) => {
+      setReplies(response.data);
+    });
+  }, [props.id]);
+
+  return (
+    <>
+      <CommentThread
```
Comment (wes-chen) [inline src/Restroom/RestroomCommentChain.js:19]: why are we doing one `CommentThread` and then a map afterwards?

**Review by wes-chen — CHANGES_REQUESTED** [review]
more comments

File: src/Restroom/RestroomCommentChain.js (hunk 1 above)
Comment (parth4apple) [reply src/Restroom/RestroomCommentChain.js]: fixed

File: src/Restroom/RestroomCommentChain.js (hunk 4 above)
Comment (parth4apple) [reply src/Restroom/RestroomCommentChain.js]: Info on rating is not available through the api without using relational aspects

File: src/Restroom/RestroomCommentChain.js (hunk 7 above)
Comment (parth4apple) [reply src/Restroom/RestroomCommentChain.js:19]: Because there is one base comment and then its replies

**Review by wes-chen — APPROVED** [review]

**Pull request merged**